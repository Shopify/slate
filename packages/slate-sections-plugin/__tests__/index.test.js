/* eslint-disable no-new */
const path = require('path');
const compiler = require('./helpers/compiler');
const SlateSectionsPlugin = require('./../');

jest.unmock('fs-extra');

test('sections with no seperate schemas, with liquid files that just need to be copied over', async () => {
  const stats = await compiler('fixtures/startersections/');

  const expectedAssetOutputKey = '../sections/test-section.liquid';
  expect(
    stats.compilation.assets[expectedAssetOutputKey]._value,
  ).toMatchSnapshot();

  const outputFiles = stats.compilation.compiler.outputFileSystem.readdirSync(
    path.resolve(__dirname, 'dist', 'sections'),
  );
  outputFiles.forEach((file) => {
    expect(
      stats.compilation.compiler.outputFileSystem.readFileSync(
        path.resolve(__dirname, 'dist', 'sections', file),
        'utf-8',
      ),
    ).toMatchSnapshot();
  });
});

test('section that has template living in folders with schema.json and no locales', async () => {
  const stats = await compiler('fixtures/seperatejsonsections/');
  const expectedAssetOutputKey = '../sections/test-section.liquid';
  expect(
    stats.compilation.assets[expectedAssetOutputKey].children[0]._value,
  ).toMatchSnapshot();
  expect(
    stats.compilation.assets[expectedAssetOutputKey].children[1]._value,
  ).toMatchSnapshot();

  const outputFiles = stats.compilation.compiler.outputFileSystem.readdirSync(
    path.resolve(__dirname, 'dist', 'sections'),
  );
  outputFiles.forEach((file) => {
    expect(
      stats.compilation.compiler.outputFileSystem.readFileSync(
        path.resolve(__dirname, 'dist', 'sections', file),
        'utf-8',
      ),
    ).toMatchSnapshot();
  });
});

test('sections that have templates living in folders with a schema.json and locales to go with aswell', async () => {
  const stats = await compiler('fixtures/normalsections/');
  // Check if file has been added to assets so webpack can output it

  const expectedAssetOutputKey = '../sections/test-section.liquid';
  expect(
    stats.compilation.assets[expectedAssetOutputKey].children[0]._value,
  ).toMatchSnapshot();

  // Check if file gets written to the output file system at the correct location, and with the correct values

  // Remove the schema tags and compare the json object to snapshot
  expect(
    JSON.parse(
      stats.compilation.assets[expectedAssetOutputKey].children[1]._value
        .replace('{% schema %}', '')
        .replace('{% endschema %}', ''),
    ),
  ).toMatchSnapshot();
});

test('throws type error if `from` key is missing from options', () => {
  const options = {
    to: path.resolve(__dirname, '../dist/sections'),
  };
  const errorToBeThrown = new TypeError('Missing or Invalid From Option');
  expect(() => {
    new SlateSectionsPlugin(options);
  }).toThrowError(errorToBeThrown);
});

test('throws type error if `from` key option is not string', () => {
  const options = {
    from: 10,
    to: path.resolve(__dirname, '../dist/sections'),
  };
  const errorToBeThrown = new TypeError('Missing or Invalid From Option');
  expect(() => {
    new SlateSectionsPlugin(options);
  }).toThrowError(errorToBeThrown);
});

test('throws type error if `to` key missing from options', () => {
  const options = {
    from: path.resolve(__dirname, '../src/sections'),
  };
  const errorToBeThrown = new TypeError('Missing or Invalid To Option');
  expect(() => {
    new SlateSectionsPlugin(options);
  }).toThrowError(errorToBeThrown);
});

test('throws type error if `to` key option is not string', () => {
  const options = {
    from: path.resolve(__dirname, '../src/sections'),
    to: 10,
  };
  const errorToBeThrown = new TypeError('Missing or Invalid To Option');
  expect(() => {
    new SlateSectionsPlugin(options);
  }).toThrowError(errorToBeThrown);
});
