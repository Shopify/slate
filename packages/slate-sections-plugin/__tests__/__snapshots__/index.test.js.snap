// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section that has template living in folders with schema.json and no locales 1`] = `
"
<section>
    <h2>{{ section.name }}</h2>
</section>

"
`;

exports[`section that has template living in folders with schema.json and no locales 2`] = `
"{% schema %}
{
  \\"name\\": \\"Test\\"
}
{% endschema %}"
`;

exports[`section that has template living in folders with schema.json and no locales 3`] = `
"
<section>
    <h2>{{ section.name }}</h2>
</section>

{% schema %}
{
  \\"name\\": \\"Test\\"
}
{% endschema %}"
`;

exports[`sections that have templates living in folders with a schema.json and locales to go with aswell 1`] = `
"<div class=\\"article\\" data-section-id=\\"{{ section.id }}\\" data-section-type=\\"article-template\\">
  <h1 class=\\"article__title h2\\">{{ article.title }}</h1>
</div>
"
`;

exports[`sections that have templates living in folders with a schema.json and locales to go with aswell 2`] = `
Object {
  "name": Object {
    "de": "Posts",
    "en": "Posts",
    "es": "Publicaciones",
    "fr": "Articles",
    "it": "Articoli",
    "ja": "投稿",
    "pt-BR": "posts",
  },
  "settings": Array [
    Object {
      "id": "image_parallax",
      "info": Object {
        "de": "Wird nur auf dem Desktop angezeigt",
        "en": "Only shows on desktop",
        "es": "Solo se muestra en el escritorio",
        "fr": "Ne s'affiche que sur le bureau",
        "it": "Appare solo su desktop",
        "ja": "デスクトップにのみ表示する",
        "pt-BR": "Aparece apenas na área de trabalho",
      },
      "label": Object {
        "de": "Parallax-Animation für Bild anzeigen",
        "en": "Show image parallax animation",
        "es": "Mostrar animación de paralaje de imagen",
        "fr": "Afficher l'animation en parallaxe",
        "it": "Mostra animazione parallasse immagine",
        "ja": "画像のパララックスアニメーションを表示する",
        "pt-BR": "Exibir animação de paralaxe da imagem",
      },
      "type": "checkbox",
    },
  ],
}
`;

exports[`sections with no seperate schemas, with liquid files that just need to be copied over 1`] = `
"
<section>
    <h2>{{ section.name }}</h2>
</section>

{% schema %}
  {
    \\"name\\": \\"Test\\",
  }
{% endschema %}
"
`;

exports[`sections with no seperate schemas, with liquid files that just need to be copied over 2`] = `
"
<section>
    <h2>{{ section.name }}</h2>
</section>

{% schema %}
  {
    \\"name\\": \\"Test\\",
  }
{% endschema %}
"
`;
